rest_framework/test.py

```diff
@@ -240,6 +244,111 @@ def request(self, **kwargs):
+                ret = renderer.render(data)
+
+                # Determine the content-type header from the renderer
+                content_type = renderer.media_type
+                if renderer.charset:
+                    content_type = "{}; charset={}".format(
+                        content_type, renderer.charset
+                    )
+
+                # Coerce text to bytes if required.
+                if isinstance(ret, str):
+                    ret = ret.encode(renderer.charset)
+
+            return ret, content_type
+
+        def get(self, path, data=None, **extra):
+            r = {
+                'QUERY_STRING': urlencode(data or {}, doseq=True),
+            }
+            if not data and '?' in path:
+                # Fix to support old behavior where you have the arguments in the
+                # url. See #1461.
+                query_string = force_bytes(path.split('?')[1])
+                query_string = query_string.decode('iso-8859-1')
+                r['QUERY_STRING'] = query_string
+            r.update(extra)
+            return self.generic('GET', path, **r)
+
+        def post(self, path, data=None, format=None, content_type=None, **extra):
+            data, content_type = self._encode_data(data, format, content_type)
+            return self.generic('POST', path, data, content_type, **extra)
+
+        def put(self, path, data=None, format=None, content_type=None, **extra):
+            data, content_type = self._encode_data(data, format, content_type)
+            return self.generic('PUT', path, data, content_type, **extra)
+
+        def patch(self, path, data=None, format=None, content_type=None, **extra):
+            data, content_type = self._encode_data(data, format, content_type)
+            return self.generic('PATCH', path, data, content_type, **extra)
+
+        def delete(self, path, data=None, format=None, content_type=None, **extra):
+            data, content_type = self._encode_data(data, format, content_type)
+            return self.generic('DELETE', path, data, content_type, **extra)
+
+        def options(self, path, data=None, format=None, content_type=None, **extra):
+            data, content_type = self._encode_data(data, format, content_type)
+            return self.generic('OPTIONS', path, data, content_type, **extra)
+
+        def generic(self, method, path, data='',
+                    content_type='application/octet-stream', secure=False, **extra):
+            # Include the CONTENT_TYPE, regardless of whether or not data is empty.
+            if content_type is not None:
+                extra['CONTENT_TYPE'] = str(content_type)
+
+            return super().generic(
+                method, path, data, content_type, secure, **extra)
+
+        def request(self, **kwargs):
+            request = super().request(**kwargs)
+            request._dont_enforce_csrf_checks = not self.enforce_csrf_checks
+            return request
```

Comment:
This may be refactored along with `APIRequestFactory` to have a common mixin to avoid having duplicated code.

Reply:
good idea, applied as suggested

Reply:
hi @ernestomancebo, I would like to have this feature (and maybe something more to handle async code also in DRF). However the final decision is up to those who owns/manage this repository, I am very interested in this project but as of now I am just a contributor.

Reply:
> I'd like to know if DRF is planning to include this contribuition

It is not, no.

You can review the conversation around this at #8496.